app/services/itemizable_update_service.rb

```diff
@@ -0,0 +1,49 @@
+module ItemizableUpdateService
+  # @param itemizable [Itemizable]
+  # @param type [Symbol] :increase or :decrease - if the original line items added quantities (purchases or
+  # donations), use :increase. If the original line_items reduced quantities (distributions) use :decrease.
+  # @param params [Hash] Parameters passed from the controller. Should include `line_item_attributes`.
+  def self.call(itemizable:, type: :increase, params: {})
+    StorageLocation.transaction do
+      item_ids = params[:line_items_attributes]&.values&.map { |i| i[:item_id].to_i } || []
+      Item.reactivate(item_ids)
+
+      from_location = to_location = itemizable.storage_location
+      to_location = StorageLocation.find(params[:storage_location_id]) if params[:storage_location_id]
+
+      increase_method = (type == :increase) ? :increase_inventory : :decrease_inventory
+      decrease_method = (type == :increase) ? :decrease_inventory : :increase_inventory
+
+      line_item_attrs = Array.wrap(params[:line_items_attributes]&.values)
+      line_item_attrs.each { |attr| attr.delete(:id) }
+
+      update_storage_location(itemizable: itemizable,
+        increase_method: increase_method,
+        decrease_method: decrease_method,
+        params: params,
+        from_location: from_location,
+        to_location: to_location)
```

Comment:
So here is what that rename would be here:

```suggestion
      update_storage_location(itemizable: itemizable,
        apply_change_method: apply_change_method,
        undo_change_method: undo_change_method,
        params: params,
        from_location: from_location,
        to_location: to_location)
```